### ISSUE TITLE
Allow optional syntax sort groups for sort-imports so single/multiple imports can be alphabetized together

### ISSUE THREAD

**Issue by Mardak** [eslint/eslint#9713]
**Tell us about your environment**

* **ESLint Version:** 4.13.1
* **Node Version:** 8.9.1
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
`--rule sort-imports:2 --env es6 --parser-options sourceType:module`

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
import a from 'foo.js';
import {b, c} from 'bar.js';
import d from 'baz.js';
```

**What did you expect to happen?**
No errors (i.e., treat single and multiple as the same when sorting)

**What actually happened? Please include the actual, raw output from ESLint.**
  2:1  error  Expected 'multiple' syntax before 'single' syntax  sort-imports

Instead of requiring "single" or "multiple" to be before or after each other, we can treat them in the same group when sorting, so they can be intermixed import syntaxes but still sorted alphabetically.

One way to fix this is to allow some "other" group that is treats everything that was not explicitly listed as the same group. E.g., `--rule sort-imports:[2,{memberSyntaxSortOrder:["other"]}]`

**Comment by ilyavolodin** [comment]
I think that `memberSyntaxSortOrder` is already complex as is, without adding additional option to it. I understand that current configuration conflicts with your style, but there's basically infinite number of ways to sort a complex list of imports, and I'm not sure we can support all of them. It's also unclear what should happen with your proposed change in the following case:
```js
import a from 'foo';
import {b, d} from 'bar';
import c from 'baz';
```

**Comment by Mardak** [comment]
The unclear example is not that different from what's currently required by the default sort-imports rule:

```js
import {b, x, y, z} from 'bar';
import {c, d} from 'baz';
import a from 'foo';
```

**Comment by Mardak** [comment]
If not adding additional options to `membebrSyntaxSortOrder`, can it at least not always require exactly 4 items? Perhaps no items up to 4 items?

**Comment by ilyavolodin** [comment]
I would need to find original issue to see why this property always requires 4 items in the list to be present. I'm sure that was discussed on the issue.

**Comment by Mardak** [comment]
The original issue for the new rule in #3143 implemented in #4820. Here's some discussion of 4 items:

https://github.com/eslint/eslint/pull/4820#discussion_r48489404
> OK, really confused about what this option if and why it must always have four options. Can you explain what this option does?

> 4 items because it describes the ordering of the related import declarations. We can rename groupByMemberSyntax to groupByMemberSyntaxOrder to give a hint what the list is about. (and of course a better documentation)

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
